internal/storage/payments.go

```diff
@@ -236,7 +236,9 @@ func (s *store) PaymentsGet(ctx context.Context, id models.PaymentID) (*models.P
 
 	status := models.PAYMENT_STATUS_PENDING
 	if len(adjustments) > 0 {
-		status = adjustments[len(adjustments)-1].Status
+		// This list is ordered by created_at DESC, so the first element is the
+		// last adjustment, and we want the last status.
+		status = adjustments[0].Status
```

Comment:
Is it consistent with our data model to have adjustments inserted in reverse chronological order, or should we be reversing the array before sending it to storage?

Reply:
Well I mean, we have `sort_id` in the payment adjustments which is auto-incremented. So we could have a situation where for some connectors display the adjustments in ASC and others in DESC chronological order by default.
But it looks like the storage layer is hardcoding the sort order in the places I could find, so most likely it is fine.